internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -49,6 +52,8 @@ var agentArtifact = artifact.Artifact{
 	Artifact: "beats/" + agentName,
 }
 
+var ErrWatcherNotStarted = errors.New("watcher did not start in time")
```

Comment:
Once you extract the timeout duration to a constant (see https://github.com/elastic/elastic-agent/pull/4229/files#r1483398147), you might as well include it in this error message so we have a bit more information while debugging.

Reply:
I chose not to put the duration in the error itself as I defined it as a specific error that can be tested with `errors.Is()` the error string should not change content depending on timeout or other.

Added a few info and error logs so humans can follow what is going on by looking at logs